server/sonar-server/src/main/java/org/sonar/server/computation/issue/SourceLinesCache.java

```diff
@@ -93,9 +98,42 @@ public String lineAuthor(@Nullable Integer lineId) {
    * Load only on demand, to avoid useless db requests on files without any new issues
    */
   private void loadIfNeeded() {
+    if (currentFileUuid == null || currentFileReportRef == null) {
+      throw new IllegalStateException(String.format("uuid (%s) and report reference (%d) must not be null to use the cache", currentFileUuid, currentFileReportRef));
+    }
     if (!loaded) {
-      dbClient.fileSourceDao().readDataStream(currentFileUuid, parserFunction);
-      loaded = true;
+      scm = reportReader.readComponentScm(currentFileReportRef);
+      loaded = scm != null;
```

Comment:
Please extract following code is another method in order to understand that at this point of the code we're gonna load data from the index